### PR TITLE
Undersampled Notebook Typo Correction

### PR DESCRIPTION
**Pull request by johannesmayer**
Evgueni detected some typos and not optimally named variables. 
The functionality of the Notebook is not affected by this, however two formulas are wrong.

### CODE REVIEW

**Comment by evgueni-ovtchinnikov** [comment]
@KrisThielemans I am afraid not.

This is what I wrote:

`C(x)` must be the square of the L2-norm of `E x – y`

CG actually finds x minimizing 
```
(A x, x) - (b, x) - (x, b) = (E x, E x) - (y, E x) - (E x, y) = (E x - y, E x - y) - (y, y)
```
Since `(y, y)` does not depend on `x`, CG essentially minimizes the norm of the square of `E x - y`.

**Comment by KrisThielemans** [comment]
@evgueni-ovtchinnikov can you review the PR and put in suggestions of how the text should be changed?

**Comment by evgueni-ovtchinnikov** [comment]
and `rr/rr0` is not the relative cost, it is (squared) relative residual

**Comment by KrisThielemans** [comment]
Both names seem correct to me. Happy to leave the current one. @evgueni-ovtchinnikov?

**Comment by evgueni-ovtchinnikov** [comment]
cost function is the square of the norm of `E x - y`, not `A x - b = E^H E x - E^H y`, which is the norm of the residual

File: notebooks/MR/d_undersampled_reconstructions.ipynb [hunk 1]
```diff
@@ -719,7 +719,7 @@
     "$$ E^H E \\, x = E^H y,$$ \n",
     "where $E$ is the encoding operator $x$ is the true image object and $y$ is the MR raw data we acquired. \n",
     "The task of image reconstruction boils down to optimizing the following function:\n",
-    "$$ \\mathcal{C}(x) = \\frac{1}{2} \\bigl{|} \\bigl{|}  E^H E \\, x - E^H y \\bigr{|} \\bigr{|}_2  \\\\\n",
+    "$$ \\mathcal{C}(x) = \\frac{1}{2} \\bigl{|} \\bigl{|}  E^H E \\, x - E^H y \\bigr{|} \\bigr{|}_2^2  \\\\\n",
```
Comment (evgueni-ovtchinnikov) [inline notebooks/MR/d_undersampled_reconstructions.ipynb]: "$$ \\mathcal{C}(x) = \\frac{1}{2} \\bigl{|} \\bigl{|}  E \\, x - y \\bigr{|} \\bigr{|}_2^2  \\\\\n"

File: notebooks/MR/d_undersampled_reconstructions.ipynb [hunk 2]
```diff
@@ -1027,12 +1027,12 @@
     "    \n",
     "    p = r + beta * p\n",
     "    \n",
-    "    cost_function = numpy.sqrt(rr/rr0)\n",
+    "    relative_cost_function = rr/rr0\n",
```
Comment (evgueni-ovtchinnikov) [inline notebooks/MR/d_undersampled_reconstructions.ipynb]: "    relative_residual = math.sqrt(rr/rr0)\n"

**Comment by evgueni-ovtchinnikov** [comment]
also, it would be nicer to compute `alpha` this way:
```
alpha = rr / Ap.dot(p) 
```

**Comment by DANAJK** [comment]
I'm not sure that you explicitly minimise a cost function in CG. Do you really mean "stopping criterion"?

**Comment by evgueni-ovtchinnikov** [comment]
In CG, the cost function is never computed explicitly (which is why Johannes got confused).

**Pull request merged**